cpp/include/raft/neighbors/detail/cagra/search_plan.cuh

```diff
@@ -69,7 +69,11 @@ struct search_plan_impl_base : public search_params {
   }
 };
 
-template <class DATA_T, class INDEX_T, class DISTANCE_T, class SAMPLE_FILTER_T>
+template <class DATA_T,
+          class INDEX_T,
+          class DISTANCE_T,
+          class DATASET_DESCRIPTOR_T,
```

Comment:
Here and in other places parameterized by the descriptor type: would it make sense to remove INDEX_T, DISTANCE_T template parameters and get them from the descriptor (i.e. `DATASET_DESCRIPTOR_T::INDEX_T`)?

Reply:
I have removed `INDEX_T`, `DATA_T`, and `DISTANCE_T` from the template params and use `DATASET_DESCRIPTOR_T::` instead.